backend/FacilityBooking/FacilitiesAPI.py

```diff
@@ -282,34 +298,77 @@ def add_booking():
             }
         ))
 
-        if (len(conflict) > 0):
-            return make_response({"err": "Conflicted booking with previous bookings.", "conflict_bookings": conflict, "status": "failed"}, 409)
-
-        if formData['facilityID'] == 15 and not db.UserCCA.find_one({'userID': formData['userID'], 'ccaID': 3}):
-            return make_response({"err": "You must be in RH Dance to make this booking", "status": "failed"}, 403)
-
         lastbookingID = list(db.Bookings.find().sort(
             [('_id', pymongo.DESCENDING)]).limit(1))
         newBookingID = 1 if len(lastbookingID) == 0 else int(
             lastbookingID[0].get("bookingID")) + 1
-
-        formData["bookingID"] = newBookingID
-
-        insertData = []
-        for i in range(formData["repeat"]):
-            insertData.append(formData.copy())
-            formData["bookingID"] += 1
-            formData["startTime"] += 7 * 24 * 60 * 60
-            formData["endTime"] += 7 * 24 * 60 * 60
-
-        db.Bookings.insert(insertData)
-
-        response = {"status": "success"}
-
+        
+        if (len(conflict) == 0):
+            formData["bookingID"] = newBookingID
+
+            insertData = []
+            for i in range(formData["repeat"]):
+                insertData.append(formData.copy())
+                formData["bookingID"] += 1
+                formData["startTime"] += 7 * 24 * 60 * 60
+                formData["endTime"] += 7 * 24 * 60 * 60
+
+            db.Bookings.insert_many(insertData)
+
+            if not formData.get("bookUntil"):
+                response = {"status": "success"}
+            else:
+                response = {"status": "success", "num_of_successful_bookings": str(len(insertData))}
+
+        elif (len(conflict) > 0):
+            if bool(formData["forceBook"]) == False:
+                return make_response({"err": "Conflicted booking with previous bookings.", "conflict_bookings": conflict, "status": "failed"}, 409)
+            else:
+                insertData = []
+                for i in range(formData["repeat"]):
+                    insertData.append(formData.copy())
+                    formData["startTime"] += 7 * 24 * 60 * 60
+                    formData["endTime"] += 7 * 24 * 60 * 60
```

Comment:
try to use global vars or abstract to helper methods to make main business logic cleaner